src/gas/gas_utils.cairo

```diff
@@ -52,29 +60,39 @@ fn pay_execution_fee(
     starting_gas: u128,
     keeper: ContractAddress,
     refund_receiver: ContractAddress
-) { // TODO
-}
+) {
+    // 63/64 gas is forwarded to external calls, reduce the startingGas to account for this
+    let reduced_starting_gas = starting_gas - sn_gasleft(array![100]) / 63;
+    let gas_used = reduced_starting_gas - sn_gasleft(array![100]);
 
-/// Pay the keeper the execution fee and refund any excess amount.
-/// # Arguments
-/// * `data_store` - The data storage contract dispatcher.
-/// * `event_emitter` - The event emitter contract dispatcher.
-/// * `bank` - The StrictBank contract holding the execution fee.
-/// * `execution_fee` - The executionFee amount.
-/// * `starting_gas` - The starting gas.
-/// * `keeper` - The keeper to pay.
-/// * `refund_receiver` - The account that should receive any excess gas refunds.
-/// # Returns
-/// * The key for the account order list.
-fn pay_execution_fee_deposit(
-    data_store: IDataStoreDispatcher,
-    event_emitter: IEventEmitterDispatcher,
-    bank: IDepositVaultDispatcher,
-    execution_fee: u128,
-    starting_gas: u128,
-    keeper: ContractAddress,
-    refund_receiver: ContractAddress
-) { // TODO
+    // each external call forwards 63/64 of the remaining gas
+    let mut execution_fee_for_keeper = adjust_gas_usage(data_store, gas_used)
+        * sn_gasprice(array![10]); //questions
+
+    if (execution_fee_for_keeper > execution_fee) {
+        execution_fee_for_keeper = execution_fee;
+    }
+
+    bank
+        .transfer_out(
+            keeper, keeper, execution_fee_for_keeper, false
+        ); // fix this, transferOutNativetoken isnt used
```

Comment:
since we aren't using `transferOutNativeToken`, i'm not 100% sure which token address we should be passing into this method. The first `keeper` param is a placeholder for now

Reply:
Let's consider eth erc20 the equivalent to "native token".  I would suggest to query its address using the function `wnt(data_store: IDataStoreDispatcher)` in token_utils. Also it might be a good initiative to rename this function to something like `fee_token` since wnt isn't the right term here.

Reply:
will do, created issue here https://github.com/keep-starknet-strange/satoru/issues/442